doc-templates/sandbox/MapboxVectorTilesApi.md

```diff
@@ -0,0 +1,191 @@
+        "type": "VehicleParking",
+        "mapper": "Stadtnavi",
+        "maxZoom": 20,
+        "minZoom": 14,
+        "cacheMaxSeconds": 60,
+        "expansionFactor": 0.25
+      },
+      // no real-time, translatable fields are translated based on accept-language header
+      // and contains less fields than the Stadtnavi mapper
+      {
+        "name": "vehicleParking",
+        "type": "VehicleParking",
+        "mapper": "Digitransit",
+        "maxZoom": 20,
+        "minZoom": 14,
+        "cacheMaxSeconds": 600,
+        "expansionFactor": 0.25
+      },
+      {
+        "name": "vehicleParkingGroups",
+        "type": "VehicleParkingGroup",
+        "mapper": "Digitransit",
+        "maxZoom": 17,
+        "minZoom": 14,
+        "cacheMaxSeconds": 600,
+        "expansionFactor": 0.25
+      }
+    ] 
+  }
+}
+```
+
+For each layer, the configuration includes:
+
+- `name` which is used in the url to fetch tiles, and as the layer name in the vector tiles.
+- `type` which tells the type of the layer. Currently supported:
+    - `Stop`
+    - `Station`
+    - `VehicleRental`: all rental places: stations and free-floating vehicles
+    - `VehicleRentalVehicle`: free-floating rental vehicles
+    - `VehicleRentalStation`: rental stations
+    - `VehicleParking`
+    - `VehicleParkingGroup`
+
+<!-- INSERT: parameters -->
+
+### Extending
+
+If more generic layers are created for this API, it should be moved out from the sandbox, into the
+core code, with potentially leaving specific property mappers in place.
+
+#### Creating a new layer
+
+In order to create a new type of layer, you need to create a new class extending `LayerBuilder<T>`.
+You need to implement two methods, `List<Geometry> getGeometries(Envelope query)`, which returns a
+list of geometries, with an object of type `T` as their userData in the geometry,
+and `double getExpansionFactor()`, which describes how much information outside the tile bounds
+should be included. This layer then needs to be added into `VectorTilesResource.layers`, with a
+new `LayerType` enum as the key, and the class constructor as the value.
+
+A new mapper needs to be added every time a new layer is added. See below for information.
+
+#### Creating a new mapper
+
+The mapping contains information of what data to include in the vector tiles. The mappers are
+defined per layer.
+
+In order to create a new mapper for a layer, you need to create a new class
+extending `PropertyMapper<T>`. In that class, you need to implement the
+method `Collection<KeyValue<String, Object>> map(T input)`. The type T is dependent on the layer for which
+you implement the mapper for. It needs to return a list of attributes, as key-value pairs which will
+be written into the vector tile.
+
+The mapper needs to be added to the `mappers` map in the layer, with a new `MapperType` enum as the
+key, and a function to create the mapper, with a `Graph` object as a parameter, as the value.
+
+## Changelog
```

Comment:
Maybe add a changelog entry. We are pretty bad in general in adding sandbox changelog entries. Maybe we should either get better at it, automate it or remove the change logs.

Reply:
I've refactored it here: https://github.com/opentripplanner/OpenTripPlanner/pull/5627/commits/cb59e3dd2d8982a2efde8d810322a3057910dcf0

Reply:
I've added it in https://github.com/opentripplanner/OpenTripPlanner/pull/5627/commits/b2e14f25fc70c5e775bb972483a5de1ea98b3fc0

I'm in favour of automating it of course. Let's discuss in the dev meeting.